Mathlib/Tactic/Bound.lean

```diff
@@ -0,0 +1,338 @@
+/-
+Copyright (c) 2024 Geoffrey Irving. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Geoffrey Irving
+-/
+
+import Aesop
+import Mathlib.Analysis.Analytic.Basic
+import Mathlib.Analysis.SpecialFunctions.Trigonometric.Basic
+import Mathlib.Analysis.SpecialFunctions.Log.Basic
+import Mathlib.Analysis.SpecialFunctions.Pow.Real
+import Mathlib.Tactic.Bound.Init
+
+/-!
+## The `bound` tactic
+
+`bound` is an `aesop` wrapper that proves inequalities by straightforward recursion on structure,
+assuming that intermediate terms are nonnegative or positive as needed.  It also has some report
+for guessing where it is unclear where to recurse, such as which side of a `min` or `max` to use
+as the bound or whether to assume a power is less than or greater than one.
+
+The functionality of `bound` overlaps with `positivity` and `gcongr`, but can jump back and forth
+between `0 ≤ x` and `x ≤ y`-type inequalities.  For example, `bound` proves
+  `0 ≤ c → b ≤ a → 0 ≤ a * c - b * c`
+By turning the goal into `b * c ≤ a * c`, then using `mul_le_mul_of_nonneg_right`.  `bound` also
+has used specialized lemmas for goals of the form `1 ≤ x, 1 < x, x ≤ 1, x < 1`.
+
+Additional hypotheses can be passed as `bound [h0, h1 n, ...]`.  This is equivalent to declaring
+them via `have` before calling `bound`.
+
+See `Ray.Tactic.BoundTests` for tests.
+
+### Calc usage
+
+Since `bound` requires the inequality proof to exactly match the structure of the expression, it is
+often useful to iterate between `bound` and `rw / simp` using `calc`.  Here is an example:
+
+```
+-- Calc example: A weak lower bound for `z ← z^2 + c`
```

Comment:
This doesn't look like the right arrow

Reply:
Sorry, it was the error for the iteration, since I was thinking about dynamics.  But that in context \mapsto is way better.  Will fix.